src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -45,24 +65,37 @@ public static async Task RetrieveClaimsAsync(LdapSettings settings, ClaimsIdenti
 
                     if (!settings.IgnoreNestedGroups)
                     {
-                        GetNestedGroups(settings.LdapConnection, identity, distinguishedName, groupCN, logger);
+                        GetNestedGroups(settings.LdapConnection, identity, distinguishedName, groupCN, logger, retrievedClaims);
                     }
                     else
                     {
-                        AddRole(identity, groupCN);
+                        retrievedClaims.Add(groupCN);
                     }
                 }
+
+                var entrySize = Encoding.Default.GetByteCount(user);
```

Comment:
I'm including the key in the entry size. Also, GetByteCount probably doesn't capture the true size in memory since there are overheads in the in-memory representation of a List<string> but I suppose it's close enough.

Reply:
Right, the size just needs to be proportional/predictable.